rust/src/main.rs

```diff
@@ -54,11 +56,23 @@ fn inner_main() -> Result<()> {
 fn main() {
     // Capture any error.  Note that in some cases the C++ code may still call exit(<code>) directly.
     let r = inner_main();
+    rpmostree_rust::ffi::rpmostree_process_global_teardown();
     // Print the error
-    if let Err(e) = r {
-        // See discussion in CxxResult for why we use this format
-        let msg = format!("{:#}", e);
-        rpmostree_rust::ffi::main_print_error(msg.as_str());
-        std::process::exit(1)
+    match r {
+        Ok(e) => std::process::exit(e),
+        Err(e) => {
+            // See discussion in CxxResult for why we use this format
+            let msg = format!("{:#}", e);
+            // Print the error message in red
+            let stderr = termcolor::BufferWriter::stderr(termcolor::ColorChoice::Auto);
+            let mut stderrbuf = stderr.buffer();
+            let _ = stderrbuf
+                .set_color(termcolor::ColorSpec::new().set_fg(Some(termcolor::Color::Red)));
+            let _ = write!(&mut stderrbuf, "error: ");
+            let _ = stderrbuf.reset();
+            let _ = writeln!(&mut stderrbuf, "{}", msg);
+            let _ = stderr.print(&stderrbuf);
```

Comment:
You may want to put the arm-body up to here into its own nested scope ending before the `exit()`, in order to give a chance to all possibly-meaningful destructors to run on the entities here (e.g. e, stderr, stderrbuf, etc).

Reply:
Sure, done.